Corona-Warn-App/src/main/java/de/rki/coronawarnapp/bugreporting/censors/BugCensor.kt

```diff
@@ -1,15 +1,44 @@
 package de.rki.coronawarnapp.bugreporting.censors
 
-import de.rki.coronawarnapp.bugreporting.debuglog.LogLine
+import kotlin.math.max
+import kotlin.math.min
 
 interface BugCensor {
 
     /**
      * If there is something to censor a new log line is returned, otherwise returns null
      */
-    suspend fun checkLog(entry: LogLine): LogLine?
+    suspend fun checkLog(message: String): CensoredString?
+
+    data class CensoredString(
+        val string: String,
+        val range: IntRange? = null
+    )
 
     companion object {
+        operator fun CensoredString.plus(newer: CensoredString?): CensoredString {
+            if (newer == null) return this
+
+            val range = when {
+                newer.range == null -> this.range
+                this.range == null -> newer.range
+                else -> min(this.range.first, newer.range.first)..max(this.range.last, newer.range.last)
+            }
+
+            return CensoredString(string = newer.string, range = range)
+        }
```

Comment:
Operator to make chaining censoring easier by being able to use "var old += new`